scripts/externalTests/runners/base.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+
+# ------------------------------------------------------------------------------
+# This file is part of solidity.
+#
+# solidity is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# solidity is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with solidity.  If not, see <http://www.gnu.org/licenses/>
+#
+# (c) 2023 solidity contributors.
+# ------------------------------------------------------------------------------
+
+import os
+import subprocess
+from abc import abstractmethod, ABCMeta
+from dataclasses import dataclass, field
+from pathlib import Path
+from shutil import rmtree
+from tempfile import mkdtemp
+from textwrap import dedent
+from typing import List, Set
+
+from test_helpers import download_project
+from test_helpers import get_solc_short_version
+from test_helpers import parse_command_line
+from test_helpers import parse_custom_presets
+from test_helpers import parse_solc_version
+from test_helpers import replace_version_pragmas
+from test_helpers import settings_from_preset
+from test_helpers import SettingsPreset
+
+CURRENT_EVM_VERSION: str = os.environ.get("DEFAULT_EVM", "shanghai")
+
+@dataclass
+class TestConfig:
+    name: str
+    repo_url: str
+    ref_type: str
+    ref: str
+    compile_only_presets: List[SettingsPreset] = field(default_factory=list)
+    settings_presets: List[SettingsPreset] = field(default_factory=lambda: list(SettingsPreset))
+    evm_version: str = field(default=CURRENT_EVM_VERSION)
+
+    def selected_presets(self) -> Set[SettingsPreset]:
+        return set(self.compile_only_presets + self.settings_presets)
+
+
+class BaseRunner(metaclass=ABCMeta):
+    config: TestConfig
+    solc_binary_type: str
+    solc_binary_path: Path
+    presets: Set[SettingsPreset]
+
+    def __init__(self, argv, config: TestConfig):
+        args = parse_command_line(f"{config.name} external tests", argv)
+        self.config = config
+        self.solc_binary_type = args.solc_binary_type
+        self.solc_binary_path = args.solc_binary_path
+        self.presets = parse_custom_presets(args.selected_presets) if args.selected_presets else config.selected_presets()
+        self.env = os.environ.copy()
+        self.tmp_dir = mkdtemp(prefix=f"ext-test-{config.name}-")
+        self.test_dir = Path(self.tmp_dir) / "ext"
+
+    def setup_solc(self) -> str:
+        if self.solc_binary_type == "solcjs":
+            # TODO: add support to solc-js
```

Comment:
This is the "TODO" I meant in https://github.com/ethereum/solidity/pull/13873#discussion_r1269595904.

Reply:
Oh, but this will be implemented. I mean, we still have tests that uses solc-js (i.e. colony). And I already have the solc-js setup done actually (I just moved to another local branch). Or we are dropping support to any external tests that uses solc-js?

Reply:
Oh, in that case it can stay.

I'd drop it eventually, because native tests are faster and this would simplify our framework, but for that we should either convert or drop colony so it can also wait.